src/plugins/dashboard/public/dashboard_api/get_dashboard_api.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import type { DashboardContainerInput } from '../../common';
+import { initializeTrackPanel } from './track_panel';
+import { initializeTrackOverlay } from './track_overlay';
+import { initializeUnsavedChanges } from './unsaved_changes';
+
+export interface InitialComponentState {
+  anyMigrationRun: boolean;
+  isEmbeddedExternally: boolean;
+  lastSavedInput: DashboardContainerInput;
+  lastSavedId: string | undefined;
+  managed: boolean;
+}
+
+export function getDashboardApi(
+  initialComponentState: InitialComponentState,
+  untilEmbeddableLoaded: (id: string) => Promise<unknown>
+) {
+  const animatePanelTransforms$ = new BehaviorSubject(false); // set panel transforms to false initially to avoid panels animating on initial render.
+  const fullScreenMode$ = new BehaviorSubject(false);
+  const managed$ = new BehaviorSubject(initialComponentState.managed);
+  const savedObjectId$ = new BehaviorSubject<string | undefined>(initialComponentState.lastSavedId);
```

Comment:
Does this need to be a behaviour subject? Does this value ever change after fetching the dashboard SO? 🤔

Reply:
`savedObjectId$` is updated on save and in `navigateToDashboard `